lib/tapioca/compilers/dsl/active_record_associations.rb

```diff
@@ -153,6 +153,10 @@ def populate_associations(mod, constant)
             add_error(<<~MSG)
               Cannot generate association `#{reflection.name}` on `#{constant}` since the source of the through association is missing.
             MSG
+          rescue NameError
```

Comment:
Would it be better to create a custom error object similar to `SourceReflectionError` when we explicitly try to load the constant instead of catching all `NameErrors` within this method? I'm afraid down the line it will hide a real `NameError` bug. But that might be too low probability to care.

Reply:
Yeah, we'd need to come up with some way of checking the actual class name without calling `reflection.klass` as that's the actual call that raises when it doesn't exist.

We can always add a ticket for this on the backlog for improvements. If other reviewers are fine with that.

Reply:
Shouldn't it be based on the `reflection.name` instead?

Something like:

```rb
raise ReflectionConstantError unless Object.const_defined?(reflection.name.to_s)
```

Reply:
Weird. `safe_constantize` seems to be causing failures locally for me, but `reflection.class_name` is working fine. Will push up the change.